kube-runtime/src/wait.rs

```diff
@@ -11,12 +11,17 @@ use crate::watcher::{self, watch_object};
 pub enum Error {
     #[error("failed to probe for whether the condition is fulfilled yet: {0}")]
     ProbeFailed(#[source] watcher::Error),
+
+    #[error("watch closed before the condition was fulfilled")]
+    WatchClosed,
```

Comment:
This should be impossible when we are sure that the user is using a `kube::runtime::watcher`, since that will always just recreate the stream if upstream closes.

Ideally we'd just forbid this entirely, but that's not supported by `Stream`. :(

Reply:
Removed this, replaced with an `expect`